### PR TITLE
API

### PR DESCRIPTION
**Pull request by carlos-ssh**
**_In this `Branch` worked on these features:_**
- Fix issues remove inf items.
- Add key for api.

[Live Version](https://carlos-ssh.github.io/librero/)

### CODE REVIEW

**Comment by omar-labana** [comment]
Hi @carlos-ssh, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, where the linters checks are not passing.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/AddBook.js [hunk 1]
```diff
@@ -2,66 +2,67 @@ import React, { useState } from 'react';
...
+
+        <a
+          href="/#"
+          className="btn-add"
+          onClick={() => {
+            add();
+          }}
+        >
+          Add Book
+        </a>
```
Comment (omar-labana) [inline src/components/AddBook.js:64]: - [ ] Adding a book through a runtime error. Also, `actions` are not correctly dispatched to the store. Please fix these issues. 
image

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,28 +1,150 @@
...
+          category: book[1][0].category,
+        }));
+
+        dispatch(loadSuccess(bookstore));
+      }
+    })
+    .catch(() => {
+      dispatch(loadFail());
+    });
+};
```
Comment (omar-labana) [inline src/redux/books/books.js:122]: - [ ] The initial request to the Bookstore API fails each time the component renders with these log errors. Kindly, update the store config (There are no logged actions and the logger library found in your package,json) then make sure actions are dispatched correctly. I'd suggest using react dev tools for a better insight on what is rendered.
image

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @carlos-ssh, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBook.js (hunk 1 above)
Comment (carlos-ssh) [reply src/components/AddBook.js:64]: I have the project working well on my desktop:
Screen Shot 2021-12-13 at 23 39 34

Comment (omar-labana) [reply src/components/AddBook.js:64]: Hey @carlos-ssh, I double-checked your project and the issue still persists, do you mind re-cloning the repo?

Comment (carlos-ssh) [reply src/components/AddBook.js:64]: Thanks for answer @omar-labana. I tried to download it last night, but i had the same issues than you!. Thanks a lot for your support and i will keep working on it to fix it ASAP. 
# 💯😄🧑‍🏭

**Review by Lameck1 — APPROVED** [review]
Hi, @carlos-ssh,

You've done well working on the previous review concerns 👏🏼
The tasks for this milestone are therefore successfully implemented 💯.
All the best with the next milestone.

You are free to merge this branch :shipit:.

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**